### DISCUSSION TITLE
Is it possible to tell which pytest plugins are loaded?

### DISCUSSION THREAD

**Discussion by kwloafman** [pytest-dev/pytest#12837] [Q&A]
I need to find out if certain pytest plugins are loaded since they cause problem and test failures: xdist, randomly, hypothesis.  I have used the `-p no:xdist` in pyproject.toml, but it's possible to bypass that.  Is there something I can put in `conftest.py` to provide a safety check so I don't go down that rabbit hole again?

**Comment by graingert** [comment]
In your conftest you can define:

```python
def pytest_configure(session)
    session.config.pluginmanager.hasplugin(name)
```
https://docs.pytest.org/en/latest/reference/reference.html#pytest.hookspec.pytest_configure
https://docs.pytest.org/en/latest/reference/reference.html#pytest.Config.pluginmanager
https://docs.pytest.org/en/latest/reference/reference.html#pytest.PytestPluginManager.hasplugin
